homeassistant/components/pterodactyl/api.py

```diff
@@ -108,10 +113,25 @@ async def async_get_data(self) -> dict[str, PterodactylData]:
                     identifier=identifier,
                     state=utilization["current_state"],
                     cpu_utilization=utilization["resources"]["cpu_absolute"],
-                    memory_utilization=utilization["resources"]["memory_bytes"],
-                    disk_utilization=utilization["resources"]["disk_bytes"],
-                    network_rx_utilization=utilization["resources"]["network_rx_bytes"],
-                    network_tx_utilization=utilization["resources"]["network_tx_bytes"],
+                    cpu_limit=server["limits"]["cpu"],
+                    memory_usage=utilization["resources"]["memory_bytes"],
+                    memory_limit=int(
+                        InformationConverter.convert(
+                            server["limits"]["memory"],
+                            UnitOfInformation.MEGABYTES,
+                            UnitOfInformation.BYTES,
+                        )
+                    ),
+                    disk_usage=utilization["resources"]["disk_bytes"],
+                    disk_limit=int(
+                        InformationConverter.convert(
+                            server["limits"]["disk"],
+                            UnitOfInformation.MEGABYTES,
+                            UnitOfInformation.BYTES,
+                        )
+                    ),
```

Comment:
We can just set the native unit of measurement to bytes and set a suggested unit of measurement

Reply:
I think I tried that before and had problems. But I'll give it a second try.

Reply:
With
```
PterodactylSensorEntityDescription(
    key=KEY_DISK_USAGE,
    translation_key=KEY_DISK_USAGE,
    value_fn=lambda data: data.disk_usage,
    entity_category=EntityCategory.DIAGNOSTIC,
    native_unit_of_measurement=UnitOfInformation.BYTES,
    suggested_unit_of_measurement=UnitOfInformation.MEGABYTES,
    suggested_display_precision=0,
),
```

I get:

> ValueError: Entity <class 'homeassistant.components.pterodactyl.sensor.PterodactylSensorEntity'> suggest an incorrect unit of measurement: MB.

Reply:
I got it now working. Was missing some other elements of the entity description.